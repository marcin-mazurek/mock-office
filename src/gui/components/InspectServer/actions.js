export const REMOVE_BUTTON_CLICKED = 'InspectServer/REMOVE_BUTTON_CLICKED';
export const removeButtonClickedAction = id => ({
  type: REMOVE_BUTTON_CLICKED,
  id
});
export const SWITCH_BUTTON_CLICKED = 'InspectServer/SWITCH_BUTTON_CLICKED';
export const switchButtonClickedAction = (id, isOn) => ({
  type: SWITCH_BUTTON_CLICKED,
  id,
  isOn
});
export const ADD_MOCK_BUTTON_CLICKED = 'InspectServer/ADD_MOCK_BUTTON_CLICKED';
export const addMockButtonClickedAction = (server, scenario, serverType) => ({
  type: ADD_MOCK_BUTTON_CLICKED,
  server,
  scenario,
  serverType
});