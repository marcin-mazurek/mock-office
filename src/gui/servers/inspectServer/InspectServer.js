import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { Scrollbars } from 'react-custom-scrollbars';
import classnames from 'classnames';
import { serverSelector } from '../../app/entities';
import FilePickerConnect from '../../mocks/importMock/FilePicker';
import MocksConnect from '../../mocks/browseMocks/Mocks';
import trashIcon from '../../../../assets/icons_gray_trash@3x.svg';
import plusIcon from '../../../../assets/icons_gray_add@3x.svg';

export const REMOVE_BUTTON_CLICKED = 'component/Server/REMOVE_BUTTON_CLICKED';
export const removeButtonClickedAction = id => ({
  type: REMOVE_BUTTON_CLICKED,
  id
});
export const SWITCH_BUTTON_CLICKED = 'component/Server/SWITCH_BUTTON_CLICKED';
export const switchButtonClickedAction = (id, isOn) => ({
  type: SWITCH_BUTTON_CLICKED,
  id,
  isOn
});
export const ADD_MOCK_BUTTON_CLICKED = 'component/Server/ADD_MOCK_BUTTON_CLICKED';
export const addMockButtonClickedAction = (server, scenario, serverType) => ({
  type: ADD_MOCK_BUTTON_CLICKED,
  server,
  scenario,
  serverType
});

const ServerToggle = ({ toggled, serverId, onSwitchButtonClick }) => {
  const toggleClassNames = classnames({
    'inspect-server-header-toggle': true,
    'inspect-server-header-toggle--up': toggled
  });

  return (
    <button
      className={toggleClassNames}
      onClick={() => onSwitchButtonClick(serverId, toggled)}
    >
      <svg className="inspect-server-header-toggle-icon" width="60px" height="60px">
        <title>icons_power/on@3x</title>
        <desc>Created with Sketch.</desc>
        <defs />
        <g id="Page-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd" opacity="0.75">
          <path
            className="inspect-server-header-toggle__icon"
            transform="translate(4, 4)"
            d="M26.0503929,17 C25.3931535,17 24.8637106,17.5364493 24.8637106,18.1941509
                    L24.8637106,23.6945939 C24.8637106,24.3559698 25.3968048,24.8887448 26.0503929,
                    24.8887448 C26.7039809,24.8887448 27.2370751,24.3522955 27.2370751,23.6945939
                    L27.2370751,18.1941509 C27.2370751,17.5364493 26.7076322,17 26.0503929,17
                    M18.0357237,27.6738723 C17.7728279,24.8262816 18.9814181,22.2469155 20.9859983
                    ,20.608173 C21.7673829,19.968843 22.9358084,20.5163152 22.9358084,21.5267506
                    C22.9358084,21.897856 22.7605446,22.2395669 22.4757409,22.4710485 C21.0553735,
                    23.6358049 20.2009623,25.4729602 20.4090881,27.5011797 C20.6792866,30.143009
                    22.7824526,32.2814577 25.4041075,32.5754026 C28.8254036,32.9612052 31.7318987,
                    30.2642612 31.7318987,26.8985927 C31.7318987,25.1128778 30.9103495,23.5145527
                    29.6286962,22.4673742 C29.3438924,22.2358926 29.1722799,21.8941817 29.1722799,
                    21.5267506 C29.1722799,20.5273382 30.3224489,19.9614943 31.0965308,20.5861271
                    C32.9294984,22.0742229 34.1052632,24.3486212 34.1052632,26.8985927 C34.1052632,
                    31.5612929 30.169093,35.3201494 25.4625287,34.9784017 C21.5702111,34.7028284
                    18.3972054,31.5833387 18.0357237,27.6738723"
            id="Fill-1" fill="#83878E"
          />
          <circle
            stroke="#83878E"
            id="Oval"
            strokeWidth="2"
            cx="28"
            cy="28"
            r="26"
            transform="translate(2, 2)"
          />
        </g>
      </svg>
    </button>
  );
};

ServerToggle.propTypes = {
  toggled: PropTypes.bool.isRequired,
  serverId: PropTypes.string.isRequired,
  onSwitchButtonClick: PropTypes.func.isRequired
};

const serverToggleMapDispatchToProps = {
  onSwitchButtonClick: switchButtonClickedAction
};

const ServerToggleConnect = connect(null, serverToggleMapDispatchToProps)(ServerToggle);

export const InspectServer = (
  {
    running,
    name,
    port,
    onRemoveButtonClick,
    scenario,
    id,
    type,
    onAddMockButtonClickedAction
  }
) =>
  <div className="inspect-server">
    <div className="inspect-server__header inspect-server-header">
      <div className="inspect-server-header__toggle">
        <ServerToggleConnect toggled={running} serverId={id} />
      </div>
      <div className="inspect-server-details">
        <div className="inspect-server-details__name">
          {name}
        </div>
        <div className="inspect-server-details__spec inspect-server-spec">
          <div className="inspect-server-spec__item">
            <span className="inspect-server-spec__label">Port:</span>
            <span className="inspect-server-spec__value">{port}</span>
          </div>
          <div className="inspect-server-spec__item">
            <span className="inspect-server-spec__label">Type:</span>
            <span className="inspect-server-spec__value">{type}</span>
          </div>
        </div>
      </div>
      <button
        className="inspect-server__remove-button button"
        onClick={() => (
          // eslint-disable-next-line no-alert
          confirm(`Do you want to stop & remove '${name}' from the list of available servers?`)
            ? onRemoveButtonClick(id)
            : false
        )
        }
      >
        <img src={trashIcon} role="presentation" />
      </button>
      <Link className="inspect-server__edit-button" to={`/server/${id}/edit`}>
        Edit
      </Link>
    </div>
    <main className="inspect-server-main inspect-server__main">
      <div className="inspect-server-mocks-header">
        <div className="inspect-server-mocks-header__label">
          Mocks:
        </div>
        <ul className="inspect-server-mocks-legend inspect-server__mocks-legend">
          <li
            className="inspect-server-mocks-legend__item
                inspect-server-mocks-legend-item"
          >
            <i
              className="inspect-server-mocks-legend__icon
                  inspect-server-mocks-legend__icon--finished"
            />
            Finished
          </li>
          <li
            className="inspect-server-mocks-legend__item
                inspect-server-mocks-legend-item"
          >
            <i
              className="inspect-server-mocks-legend__icon
                  inspect-server-mocks-legend__icon--in-progress"
            />
            In progress
          </li>
          <li
            className="inspect-server-mocks-legend__item
                inspect-server-mocks-legend-item"
          >
            <i
              className="inspect-server-mocks-legend__icon
                  inspect-server-mocks-legend__icon--cancelled"
            />
            Cancelled
          </li>
          <li
            className="inspect-server-mocks-legend__item
                inspect-server-mocks-legend-item"
          >
            <i className="inspect-server-mocks-legend__icon" />Ready
          </li>
        </ul>
        <button
          className="inspect-server__add-mock-button inspect-server-mocks-header__button"
          onClick={() => onAddMockButtonClickedAction(id, scenario, type)}
        >
          <img src={plusIcon} role="presentation" style={{ marginRight: '11px' }} />
          Add mock
        </button>
        <FilePickerConnect scenario={scenario} server={id} />
      </div>
      <div className="inspect-server__mocks">
        <div className="inspect-server__mocks-scroll-container">
          <Scrollbars>
            <MocksConnect server={id} scenario={scenario} />
          </Scrollbars>
        </div>
      </div>
    </main>
  </div>;

InspectServer.propTypes = {
  running: PropTypes.bool.isRequired,
  onRemoveButtonClick: PropTypes.func.isRequired,
  onAddMockButtonClickedAction: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  port: PropTypes.number.isRequired,
  scenario: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired
};

const serverMapDispatchToProps = {
  onRemoveButtonClick: removeButtonClickedAction,
  onAddMockButtonClickedAction: addMockButtonClickedAction
};

const serverMapStateToProps = (state, ownProps) => {
  const server = serverSelector(state, ownProps.params.id);

  return {
    id: server.id,
    type: server.type,
    name: server.name,
    port: server.port,
    scenario: server.scenario,
    running: server.running
  };
};

export default connect(
  serverMapStateToProps, serverMapDispatchToProps
)(InspectServer);
