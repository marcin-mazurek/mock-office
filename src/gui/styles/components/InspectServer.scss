$padding-left: 24px;

.inspect-server {
  display: flex;
  flex: auto;
  flex-direction: column;
  overflow-y: auto;
}

.inspect-server__header {
  flex: none;
}

.inspect-server-details {
  margin-right: auto;
}

.inspect-server-details__name {
  font-weight: 500;
  font-size: 24px;
  margin-bottom: 6px;
}

.inspect-server-header {
  display: flex;
  padding: 24px 24px 24px $padding-left;
  align-items: center;
  border-bottom: 1px solid #EBEBEB;
}

.inspect-server-spec {
  display: flex;
  font-size: 12px;
  line-height: 14px;
  letter-spacing: 0.5px;
}

.inspect-server-spec__item {
  color: #60666F;
  margin-right: 12px;
}

.inspect-server-spec__value {
  margin-left: 3px;
}

.inspect-server-spec__label {
  opacity: 0.39;
  font-weight: 300;
}

.inspect-server__main {
  flex: auto;
  background-color: #F6F7F9;
}

.inspect-server-main {
  display: flex;
  flex-direction: column;
}

.inspect-server__add-scene-button {
  @extend %button-reset;
  text-decoration: none;
  margin-right: 10px;
  padding: 0 20px;
  border-right: 1px solid #D8D8D8;
}

$size: 40px;

.inspect-server-header-toggle__icon {
  transition: all .5s ease;
}

.inspect-server-header-toggle {
  margin-right: 20px;
  cursor: pointer;

  #Oval {
    transition: all .3s ease;
  }

  &:hover {
    #Oval {
      r: 28;
    }
  }
}

.inspect-server-header-toggle--up {
  #Oval {
    stroke: #23D1A9;
    r: 28;
  }

  .inspect-server-header-toggle__icon {
    fill: #23D1A9;
  }
}

.inspect-server-header-toggle {
  @extend %button-reset;
  display: flex;
  align-items: center;
  justify-content: center;
  transition: all .1s ease-in-out;
  padding: 0;
}

.inspect-server__scenes {
  flex: auto;
  position: relative;
}

.inspect-server-scenes-header {
  color: #727880;
  padding: 16px 15px 16px $padding-left;
  font-weight: 500;
  display: flex;
  align-items: center;
  flex: none;
  font-size: 15px;
}

.inspect-server-scenes-header__button {
  &:hover {
    color: #23D1A9;
  }
}

.inspect-server-scenes-header__label {
  margin-right: 50px;
}

.inspect-server__scenes-legend {
  margin-right: auto;
}

.inspect-server-scenes-legend {
  @extend %list-reset;
  display: flex;
  font-weight: 300;
  color: #6B717A;
  font-size: 12px;
}

.inspect-server-scenes-legend__item {
  margin-right: 18px;
}

.inspect-server-scenes-legend-item {
  display: flex;
  align-items: center;
}

.inspect-server-scenes-legend__icon {
  display: inline-block;
  border-radius: 100%;
  height: 6px;
  width: 6px;
  background-color: #E1E5EE;
  margin-right: 8px;
}

.inspect-server-scenes-legend__icon--cancelled {
  background-color: #EC9292;
}

.inspect-server-scenes-legend__icon--in-progress {
  background-color: #B9EBE0;
}

.inspect-server-scenes-legend__icon--finished {
  background-color: #23D1A9;
}

.inspect-server__scenes-scroll-container {
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
}

.inspect-server__remove-button {
  background-color: transparent;
  font-size: 24px;
  transition: color .2s ease;
  cursor: pointer;

  &:hover {
    opacity: .5;
  }
}

.inspect-server__edit-button {
  text-decoration: none;
  margin: 0 10px;
}
