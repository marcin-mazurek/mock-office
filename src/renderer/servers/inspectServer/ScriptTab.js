import React from 'react';
import { connect } from 'react-redux';
import { getSelectedServerScripts } from '../../serverScripts/selectors';
import { getSelected } from '../selectors';
import ScriptPicker from '../../serverScripts/addFromFile/FilePicker';
import ScriptTrigger from './ScriptTrigger';

const ScriptTab = ({ scripts, server }) => (
  <div className="server-script-tab">
    <ScriptPicker />
    <div>
      <h3>scripts:</h3>
      <ul>
        {
          scripts.map(script => (
            <li key={script.id}>
              <ScriptTrigger script={script} server={server} />
            </li>
          ))
        }
      </ul>
    </div>
  </div>
);

ScriptTab.propTypes = {
  scripts: React.PropTypes.shape({}),
  server: React.PropTypes.string
};

const mapStateToProps = state => ({
  server: getSelected(state),
  scripts: getSelectedServerScripts(state)
});

export default connect(mapStateToProps)(ScriptTab);
