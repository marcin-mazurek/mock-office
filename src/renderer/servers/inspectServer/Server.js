import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import StartButtonConnect from '../startServer/StartButton';
import StopButtonConnect from '../stopServer/StopButton';
import { isRunning, getSelectedServerDetails, getSelected, getAllAsList } from '../selectors';
import FilePickerConnect from '../../tasks/addTaskFromFile/FilePicker';
import Queue from '../../queues/Queue';

export const ServerPlaceholder = ({ serverExists }) =>
  <div className="server-placeholder">
    {serverExists ? 'Select server' : 'Add server'}
  </div>;

ServerPlaceholder.propTypes = {
  serverExists: React.PropTypes.bool.isRequired
};

const serverPlaceholderMapStateToProps = state => ({
  serverExists: !getAllAsList(state).isEmpty()
});

const ServerPlaceholderConnect = connect(serverPlaceholderMapStateToProps)(ServerPlaceholder);

export const ServerInspect = ({ running, serverDetails }) => {
  const { name, port, type, id } = serverDetails;

  return (
    <div className="inspect-server">
      <div className="view-header">
        { running ? <StopButtonConnect /> : <StartButtonConnect /> }
        {name}
      </div>
      <div className="inspect-server-spec-bar">
        <div className="inspect-server-spec-bar__cell">
          Port:
          <span className="inspect-server-spec-bar__value">{port}</span>
        </div>
        <div className="inspect-server-spec-bar__cell">
          Type:
          <span className="inspect-server-spec-bar__value">{type}</span>
        </div>
      </div>
      <main className="inspect-server-main inspect-server__main">
        <div className="inspect-server-details">
          <div className="inspect-server-details__row">
            <div className="inspect-server-details__cell">
              <Link to="/add-task" className="button">
                <i className="fa fa-plus" /> Add task
              </Link>
            </div>
            <div className="inspect-server-details__cell">
              Add from file: <FilePickerConnect serverId={id} />
            </div>
          </div>
        </div>
        <div className="inspect-server-queue">
          <Queue id={id} />
        </div>
      </main>
    </div>
  );
};

ServerInspect.propTypes = {
  running: React.PropTypes.bool.isRequired,
  serverDetails: React.PropTypes.shape({})
};

export const Server = ({ selected, running, serverDetails }) => (
  selected
    ? <ServerInspect running={running} serverDetails={serverDetails} />
    : <ServerPlaceholderConnect />
);

Server.propTypes = {
  selected: React.PropTypes.string,
  running: React.PropTypes.bool.isRequired,
  serverDetails: React.PropTypes.shape({})
};

const serverMapStateToProps = state => ({
  selected: getSelected(state),
  running: isRunning(state),
  serverDetails: getSelectedServerDetails(state)
});

export default connect(serverMapStateToProps)(Server);
