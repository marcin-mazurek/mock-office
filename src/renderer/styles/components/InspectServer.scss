.inspect-server {
  display: flex;
  flex: auto;
  flex-direction: column;
  overflow-y: auto;
}

.inspect-server__header {
  flex: none;
}

.inspect-server-details {
  margin-right: auto;
}

.inspect-server-details__name {
  font-size: 26px;
  font-weight: 500;
}

.inspect-server-header {
  display: flex;
  padding: 10px 15px;
  align-items: center;
  border-bottom: $inspect-server-header-border-bottom;
}

.inspect-server-spec {
  display: flex;
  font-size: 12px;
  letter-spacing: 0.5px;
}

.inspect-server-spec__item {
  padding: 4px 7px;
  border-right: 1px solid #EEE;

  &:first-child {
    padding-left: 0;
  }

  &:last-child {
    border-right: none;
  }
}

.inspect-server-spec__value {
  margin-left: 3px;
}

.inspect-server-spec__label {
  color: $inspect-server-spec-label-font-color;
  font-weight: 300;
}

.inspect-server__main {
  flex: auto;
}

.inspect-server-main {
  display: flex;
  flex-direction: column;
}

.inspect-server__add-task-button {
  border: $inspect-server-add-task-button-border;
  margin-right: 10px;
}

$size: 40px;

.inspect-server-header__toggle {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 50px;
  margin-right: 10px;
}

.inspect-server-header-toggle {
  background-color: $start-button-background-color;
  font-size: 21px;
  border-radius: 100%;
  cursor: pointer;
  height: $size;
  width: $size;
  display: flex;
  align-items: center;
  justify-content: center;
  border: $start-button-border;
  border: $start-button-default-border;
  transition: all .1s ease-in-out;

  &:hover {
    color: $start-button-font-color;
    border: $start-button-border;
    height: $size + 5;
    width: $size + 5;
  }
}

.inspect-server-header-toggle.inspect-server-header-toggle--up {
  color: $start-button-font-color;
  border: $start-button-border;

  &:hover {
    color: $stop-button-font-color;
    border: 2px solid $stop-button-font-color;
  }
}

.inspect-server__tasks {
  flex: auto;
  position: relative;
}

.inspect-server-tasks-header {
  background-color: $tasks-header-background-color;
  padding: 10px 15px;
  font-weight: 500;
  display: flex;
  align-items: center;
  flex: none;
}

.inspect-server-tasks-header__button {
  &:hover {
    color: $primary-color;
  }
}

.inspect-server-tasks-header__label {
  margin-right: auto;
}

.inspect-server__tasks-scroll-container {
  height: 100%;
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
}

.inspect-server__remove-button {
  background-color: transparent;
  font-size: 24px;
  transition: color .2s ease;
  cursor: pointer;
  color: $inspect-server-remove-button-font-color;

  &:hover {
    color: $inspect-server-remove-button-hover-font-color;
  }
}

