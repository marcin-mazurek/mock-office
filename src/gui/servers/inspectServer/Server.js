import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { Scrollbars } from 'react-custom-scrollbars';
import classnames from 'classnames';
import createStartAction from '../startServer/actions';
import createStopAction from '../stopServer/actions';
import {
  isRunning,
  getServerDetails
} from '../../entities/servers/selectors';
import FilePickerConnect from '../../scenes/addSceneFromFile/FilePicker';
import ScenesConnect from '../../scenes/browseScenes/Scenes';
import { init as createRemoveServerAction } from '../removeServer/actions';
import { init as createRenameServerAction } from '../renameServer/actions';
import trashIcon from '../../assets/icons_gray_trash@3x.svg';
import plusIcon from '../../assets/icons_gray_add@3x.svg';

const ServerToggle = ({ toggled, serverId, stop, start }) => {
  let handleClick;

  if (toggled) {
    handleClick = () => {
      stop(serverId);
    };
  } else {
    handleClick = () => {
      start(serverId);
    };
  }

  const toggleClassNames = classnames({
    'inspect-server-header-toggle': true,
    'inspect-server-header-toggle--up': toggled
  });

  return (
    <button
      className={toggleClassNames}
      onClick={handleClick}
    >
      <svg className="inspect-server-header-toggle-icon" width="60px" height="60px">
        <title>icons_power/on@3x</title>
        <desc>Created with Sketch.</desc>
        <defs />
        <g id="Page-1" stroke="none" strokeWidth="1" fill="none" fillRule="evenodd" opacity="0.75">
          <path
            className="inspect-server-header-toggle__icon"
            transform="translate(4, 4)"
            d="M26.0503929,17 C25.3931535,17 24.8637106,17.5364493 24.8637106,18.1941509
                    L24.8637106,23.6945939 C24.8637106,24.3559698 25.3968048,24.8887448 26.0503929,
                    24.8887448 C26.7039809,24.8887448 27.2370751,24.3522955 27.2370751,23.6945939
                    L27.2370751,18.1941509 C27.2370751,17.5364493 26.7076322,17 26.0503929,17
                    M18.0357237,27.6738723 C17.7728279,24.8262816 18.9814181,22.2469155 20.9859983
                    ,20.608173 C21.7673829,19.968843 22.9358084,20.5163152 22.9358084,21.5267506
                    C22.9358084,21.897856 22.7605446,22.2395669 22.4757409,22.4710485 C21.0553735,
                    23.6358049 20.2009623,25.4729602 20.4090881,27.5011797 C20.6792866,30.143009
                    22.7824526,32.2814577 25.4041075,32.5754026 C28.8254036,32.9612052 31.7318987,
                    30.2642612 31.7318987,26.8985927 C31.7318987,25.1128778 30.9103495,23.5145527
                    29.6286962,22.4673742 C29.3438924,22.2358926 29.1722799,21.8941817 29.1722799,
                    21.5267506 C29.1722799,20.5273382 30.3224489,19.9614943 31.0965308,20.5861271
                    C32.9294984,22.0742229 34.1052632,24.3486212 34.1052632,26.8985927 C34.1052632,
                    31.5612929 30.169093,35.3201494 25.4625287,34.9784017 C21.5702111,34.7028284
                    18.3972054,31.5833387 18.0357237,27.6738723"
            id="Fill-1" fill="#83878E"
          />
          <circle
            stroke="#83878E"
            id="Oval"
            strokeWidth="2"
            cx="28"
            cy="28"
            r="26"
            transform="translate(2, 2)"
          />
        </g>
      </svg>
    </button>
  );
};

ServerToggle.propTypes = {
  toggled: PropTypes.bool.isRequired,
  serverId: PropTypes.string.isRequired,
  stop: PropTypes.func.isRequired,
  start: PropTypes.func.isRequired
};

const serverToggleMapDispatchToProps = {
  start: createStartAction,
  stop: createStopAction
};

const ServerToggleConnect = connect(null, serverToggleMapDispatchToProps)(ServerToggle);

export const Server = ({ running, serverDetails, removeServer }) => {
  const { name, port, type, id } = serverDetails;

  return (
    <div className="inspect-server">
      <div className="inspect-server__header inspect-server-header">
        <div className="inspect-server-header__toggle">
          <ServerToggleConnect toggled={running} serverId={id} />
        </div>
        <div className="inspect-server-details">
          <div className="inspect-server-details__name">
            {name}
          </div>
          <div className="inspect-server-details__spec inspect-server-spec">
            <div className="inspect-server-spec__item">
              <span className="inspect-server-spec__label">Port:</span>
              <span className="inspect-server-spec__value">{port}</span>
            </div>
            <div className="inspect-server-spec__item">
              <span className="inspect-server-spec__label">Type:</span>
              <span className="inspect-server-spec__value">{type}</span>
            </div>
          </div>
        </div>
        <button
          className="inspect-server__remove-button button"
          onClick={() => (
            // eslint-disable-next-line no-alert
            confirm(`Do you want to stop & remove '${name}' from the list of available servers?`)
              ? removeServer(id)
              : false
          )
          }
        >
          <img src={trashIcon} role="presentation" />
        </button>
      </div>
      <main className="inspect-server-main inspect-server__main">
        <div className="inspect-server-scenes-header">
          <div className="inspect-server-scenes-header__label">
            Scenes:
          </div>
          <ul className="inspect-server-scenes-legend inspect-server__scenes-legend">
            <li
              className="inspect-server-scenes-legend__item
              inspect-server-scenes-legend-item"
            >
              <i
                className="inspect-server-scenes-legend__icon
                inspect-server-scenes-legend__icon--finished"
              />
              Finished
            </li>
            <li
              className="inspect-server-scenes-legend__item
              inspect-server-scenes-legend-item"
            >
              <i
                className="inspect-server-scenes-legend__icon
                inspect-server-scenes-legend__icon--in-progress"
              />
              In progress
            </li>
            <li
              className="inspect-server-scenes-legend__item
              inspect-server-scenes-legend-item"
            >
              <i
                className="inspect-server-scenes-legend__icon
                inspect-server-scenes-legend__icon--cancelled"
              />
              Cancelled
            </li>
            <li
              className="inspect-server-scenes-legend__item
              inspect-server-scenes-legend-item"
            >
              <i className="inspect-server-scenes-legend__icon" />Ready
            </li>
          </ul>
          <Link
            to={`${id}/add-scene`}
            className="inspect-server__add-scene-button inspect-server-scenes-header__button"
          >
            <img src={plusIcon} role="presentation" style={{ marginRight: '11px' }} />
            Add scene
          </Link>
          <FilePickerConnect serverId={id} />
        </div>
        <div className="inspect-server__scenes">
          <div className="inspect-server__scenes-scroll-container">
            <Scrollbars>
              <ScenesConnect serverId={id} />
            </Scrollbars>
          </div>
        </div>
      </main>
    </div>
  );
};

Server.propTypes = {
  running: React.PropTypes.bool.isRequired,
  serverDetails: React.PropTypes.shape({}),
  removeServer: React.PropTypes.func.isRequired
};

const serverMapDispatchToProps = {
  removeServer: createRemoveServerAction,
  serverNameChange: createRenameServerAction
};

const serverMapStateToProps = (state, ownProps) => ({
  running: isRunning(state, ownProps.params.id),
  serverDetails: getServerDetails(state, ownProps.params.id)
});

export default connect(
  serverMapStateToProps, serverMapDispatchToProps
)(Server);
