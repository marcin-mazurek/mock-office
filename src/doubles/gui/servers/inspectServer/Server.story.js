import React from 'react';
import { storiesOf } from '@kadira/storybook';
import { ServerPlaceholder } from './Server';

storiesOf('ServerPlaceholder', module)
  .add('server exists', () => (
    <div style={{ height: 500, width: 500 }}>
      <ServerPlaceholder serverExists />
    </div>
  ))
  .add('any server exists', () => (
    <div style={{ height: 500, width: 500 }}>
      <ServerPlaceholder serverExists={false} />
    </div>
  ));
