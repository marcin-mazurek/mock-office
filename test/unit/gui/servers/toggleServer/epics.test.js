import { switchButtonClickedAction } from '../../../../../src/gui/servers/inspectServer/InspectServer';

/* eslint-disable global-require */
describe('toggleServerEpic', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  describe('when server is down', () => {
    test('actions snapshot when response object hasn\'t got errors', (done) => {
      const configureMockStore = require('redux-mock-store').default;
      const createEpicMiddleware = require('redux-observable').createEpicMiddleware;
      jest.mock('../../../../../src/gui/resources/api', () => ({
        startServer() {
          return Promise.resolve({
            data: { id: 'some id' }
          });
        }
      }));
      // eslint-disable-next-line global-require
      const toggleServerEpic = require('../../../../../src/gui/servers/toggleServer/epic').default;

      const epicMiddleware = createEpicMiddleware(toggleServerEpic);
      const mockStore = configureMockStore([epicMiddleware]);
      const store = mockStore();
      store.dispatch(switchButtonClickedAction('some id', false));

      Promise.resolve().then(() => {
        expect(store.getActions()).toMatchSnapshot();
        done();
      });
    });

    test('actions snapshot when on api request errors', (done) => {
      const configureMockStore = require('redux-mock-store').default;
      const createEpicMiddleware = require('redux-observable').createEpicMiddleware;
      jest.mock('../../../../../src/gui/resources/api', () => ({
        startServer() {
          return Promise.resolve({ error: 'error message' });
        }
      }));
      const toggleServerEpic = require('../../../../../src/gui/servers/toggleServer/epic').default;

      const epicMiddleware = createEpicMiddleware(toggleServerEpic);
      const mockStore = configureMockStore([epicMiddleware]);
      const store = mockStore();
      store.dispatch(switchButtonClickedAction('some id', false));

      Promise.resolve().then(() => {
        expect(store.getActions()).toMatchSnapshot();
        done();
      });
    });
  });

  describe('when server is up', () => {
    test('actions snapshot when response object hasn\'t got errors', (done) => {
      const configureMockStore = require('redux-mock-store').default;
      const createEpicMiddleware = require('redux-observable').createEpicMiddleware;
      jest.mock('../../../../../src/gui/resources/api', () => ({
        stopServer() {
          return Promise.resolve({
            data: { id: 'some id' }
          });
        }
      }));
      // eslint-disable-next-line global-require
      const toggleServerEpic = require('../../../../../src/gui/servers/toggleServer/epic').default;

      const epicMiddleware = createEpicMiddleware(toggleServerEpic);
      const mockStore = configureMockStore([epicMiddleware]);
      const store = mockStore();
      store.dispatch(switchButtonClickedAction('some id', false));

      Promise.resolve().then(() => {
        expect(store.getActions()).toMatchSnapshot();
        done();
      });
    });

    test('actions snapshot when on api request errors', (done) => {
      const configureMockStore = require('redux-mock-store').default;
      const createEpicMiddleware = require('redux-observable').createEpicMiddleware;
      jest.mock('../../../../../src/gui/resources/api', () => ({
        startServer() {
          return Promise.resolve({ error: 'error message' });
        }
      }));
      const toggleServerEpic = require('../../../../../src/gui/servers/toggleServer/epic').default;

      const epicMiddleware = createEpicMiddleware(toggleServerEpic);
      const mockStore = configureMockStore([epicMiddleware]);
      const store = mockStore();
      store.dispatch(switchButtonClickedAction('some id', false));

      Promise.resolve().then(() => {
        expect(store.getActions()).toMatchSnapshot();
        done();
      });
    });
  });
});
