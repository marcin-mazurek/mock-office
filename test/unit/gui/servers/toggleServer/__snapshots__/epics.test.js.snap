// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggleServerEpic when server is down actions snapshot when on api request errors 1`] = `
Array [
  Object {
    "id": "some id",
    "isOn": false,
    "type": "InspectServer/SWITCH_BUTTON_CLICKED",
  },
  Object {
    "reason": "error message",
    "type": "toggleServer/FAILED",
  },
]
`;

exports[`toggleServerEpic when server is down actions snapshot when response object hasn't got errors 1`] = `
Array [
  Object {
    "id": "some id",
    "isOn": false,
    "type": "InspectServer/SWITCH_BUTTON_CLICKED",
  },
  Object {
    "id": "some id",
    "type": "toggleServer/SUCCEEDED",
  },
]
`;

exports[`toggleServerEpic when server is up actions snapshot when on api request errors 1`] = `
Array [
  Object {
    "id": "some id",
    "isOn": false,
    "type": "InspectServer/SWITCH_BUTTON_CLICKED",
  },
  Object {
    "reason": "error message",
    "type": "toggleServer/FAILED",
  },
]
`;

exports[`toggleServerEpic when server is up actions snapshot when response object hasn't got errors 1`] = `
Array [
  Object {
    "id": "some id",
    "isOn": false,
    "type": "InspectServer/SWITCH_BUTTON_CLICKED",
  },
]
`;
